include/ginkgo/core/matrix/csr.hpp

```diff
@@ -35,6 +35,7 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 
 #include <ginkgo/core/base/array.hpp>
+#include <ginkgo/core/base/index_set.hpp>
```

Comment:
Yeah, I feel an include where possible is a much cleaner way to go rather than having incomplete types, especially when we know that the class/file we are including is a basic data structure and is guaranteed to have no dependence on the class it is being included in.